lib/sensors/temperature.cpp

```diff
@@ -15,33 +32,37 @@ Temperature::~Temperature()
 {
 }
 
-std::optional<std::int16_t> Temperature::read()
+std::optional<core::Result> Temperature::isValueReady()
 {
   const auto status_check_result = i2c_->readByte(kTemperatureDefaultAddress, kStatus);
   if (!status_check_result) {
     logger_.log(
       core::LogLevel::kFatal, "Failed to read temperature sensor status at channel %d", channel_);
     return std::nullopt;
   }
-  // TODOLater: change this from a fatal error to a debug and re-prompt read (with ROS)
   if (status_check_result.value() == kBusy) {
-    logger_.log(core::LogLevel::kFatal,
+    logger_.log(core::LogLevel::kWarn,
                 "Failed to read, temperature sensor is not ready to be read at channel %d",
                 channel_);
-    return std::nullopt;
+    return core::Result::kFailure;
   }
   if (status_check_result.value() == kTemperatureOverUpperLimit) {
-    logger_.log(core::LogLevel::kFatal,
+    logger_.log(core::LogLevel::kWarn,
                 "Failed to read, temperature is above upper limit at channel %d",
                 channel_);
```

Comment:
what does being over upper or below lower limits mean? is it a temporary thing or not? if not then the return should distinguish between these and not ready

Reply:
not sure since the datasheet was a bit vague here, though you are right, I'll make this an fatal error